python/cuml/cuml/dask/neighbors/kneighbors_classifier.py

```diff
@@ -316,9 +316,12 @@ def score(self, X, y, convert_dtype=True):
         -------
         score
         """
-        y_pred = self.predict(X, convert_dtype=convert_dtype)
-        if not isinstance(y_pred, da.Array):
-            y_pred = y_pred.to_dask_array(lengths=True)
+        y_pred_plain = self.predict(X, convert_dtype=convert_dtype)
```

Comment:
This variable needs to be alive for at least as long as the `mean_match.compute()` down below.